Windows jobs on Jenkins hang on sbt startup since sbt 1.3 upgrade

both 2.12.x and 2.13.x are affected, so we should fix on 2.12.x and merge the fix forwards.

* 2.12: https://github.com/scala/scala/pull/8660, https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-windows/
* 2.13: https://github.com/scala/scala/pull/8525, https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.13.x-integrate-windows/

https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-windows/1642/console

```
++ SBT='java -Dfile.encoding=UTF-8 -server -XX:+AggressiveOpts -XX:+UseParNewGC -Xmx2G -Xss1M -XX:MaxPermSize=512M -XX:ReservedCodeCacheSize=128M -Dpartest.threads=4 -Duser.home=y:/jenkins -Djava.io.tmpdir=y:/jenkins/tmp -Dhudson.remoting.RemoteClassLoader.force=com.sun.jna.Native -Dsbt.ivy.home=y:\jenkins\workspace\scala-2.12.x-integrate-windows/.ivy2 -jar C:\sbt\sbt-launch.jar -Dsbt.override.build.repos=true -Dsbt.repository.config=y:\jenkins\workspace\scala-2.12.x-integrate-windows/scripts/sbt-repositories-config'
++ java -Dfile.encoding=UTF-8 -server -XX:+AggressiveOpts -XX:+UseParNewGC -Xmx2G -Xss1M -XX:MaxPermSize=512M -XX:ReservedCodeCacheSize=128M -Dpartest.threads=4 -Duser.home=y:/jenkins -Djava.io.tmpdir=y:/jenkins/tmp -Dhudson.remoting.RemoteClassLoader.force=com.sun.jna.Native '-Dsbt.ivy.home=y:\jenkins\workspace\scala-2.12.x-integrate-windows/.ivy2' -jar 'C:\sbt\sbt-launch.jar' -Dsbt.override.build.repos=true '-Dsbt.repository.config=y:\jenkins\workspace\scala-2.12.x-integrate-windows/scripts/sbt-repositories-config' -warn setupPublishCore generateBuildCharacterPropertiesFile publishLocal
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? Build timed out (after 200 minutes). Marking the build as aborted.
Build was aborted
```
as originally reported a couple weeks ago [here](https://github.com/scala/scala/pull/8660#issuecomment-578337725), but we merged the upgrades anyway, which is fine. this has now worked its way to the top of my priorities

the jobs run `scripts/jobs/integrate/windows` — one place to start would be to add `-debug` to the sbt invocation (on a branch) to get more info about where it's getting stuck

okay, now at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-windows/1644/console we see:

```
[error] Caused by: java.io.IOException: Cannot run program "powershell.exe": CreateProcess error=2, The system cannot find the file specified
[error] 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
[error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.Util.runCommands(Util.java:176)
[error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.Util.getWinDirs(Util.java:147)
[error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.Util.getWinDirs(Util.java:120)
[error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.ProjectDirectories.fromPath(ProjectDirectories.java:221)
[error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.ProjectDirectories.from(ProjectDirectories.java:274)
[error] 	at lmcoursier.internal.shaded.coursier.paths.CoursierPaths.coursierDirectories(CoursierPaths.java:82)
[error] 	at lmcoursier.internal.shaded.coursier.paths.CoursierPaths.computeCacheDirectory(CoursierPaths.java:44)
[error] 	at lmcoursier.internal.shaded.coursier.paths.CoursierPaths.cacheDirectory(CoursierPaths.java:70)
[error] 	at lmcoursier.internal.shaded.coursier.paths.CachePath.defaultCacheDirectory(CachePath.java:93)
[error] 	at lmcoursier.internal.shaded.coursier.cache.CacheDefaults$.location$lzycompute(CacheDefaults.scala:16)
[error] 	at lmcoursier.internal.shaded.coursier.cache.CacheDefaults$.location(CacheDefaults.scala:16)
[error] 	at lmcoursier.internal.shaded.coursier.cache.FileCache$.apply(FileCache.scala:1088)
[error] 	at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:134)
[error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:60)
```

this appears to be https://github.com/soc/directories-jvm/issues/29 (to which https://github.com/sbt/sbt/issues/5386 is a reference)

I think the next step here is probably for me to get ssh access to the Windows worker, otherwise the troubleshooting process (pushing changes to `scripts/jobs/integrate/windows`) is just too cumbersome

--- a/scripts/jobs/integrate/windows
+++ b/scripts/jobs/integrate/windows
@@ -7,7 +7,12 @@ javac -version
 
 generateRepositoriesConfig
 
-SBT="java $JAVA_OPTS -Dsbt.ivy.home=$WORKSPACE/.ivy2 -jar $sbtLauncher -Dsbt.override.build.repos=true -Dsbt.repository.config=$sbtRepositoryConfig"
+# it may not be necessary to set both COURSIER_HOME and sbt.coursier.home,
+# but at least for now, doing it just in case; see discussion at
+# https://github.com/scala/scala-dev/issues/666
+export COURSIER_HOME=$WORKSPACE/.coursier
+
+SBT="java $JAVA_OPTS -Dsbt.ivy.home=$WORKSPACE/.ivy2 -Dsbt.coursier.home=$WORKSPACE/.coursier -jar $sbtLauncher -Dsbt.override.build.repos=true -Dsbt.repository.config=$sbtRepositoryConfig"
 
 # Build locker with STARR
 $SBT -warn "setupPublishCore" generateBuildCharacterPropertiesFile publishLocal

